patches/0038-ipq40xx-Add-support-for-CIG-WF610D-AP.patch

```diff
@@ -0,0 +1,639 @@
+@@ -69,6 +69,11 @@ ipq40xx_setup_interfaces()
+ 		ucidef_add_switch "switch0" \
+ 			"0u@eth0" "3:lan" "4:lan" "0u@eth1" "5:wan"
+ 		;;
++	cig,wf610d)
++                ucidef_set_interface_wan "eth0"
++                ucidef_set_interface_lan ""
++                ;;
++
+ 	*)
+ 		echo "Unsupported hardware. Network interfaces not initialized"
+ 		;;
+diff --git a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+index ff77aed304..021a4965ec 100644
+--- a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
++++ b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+@@ -169,7 +169,8 @@ case "$FIRMWARE" in
+ 	openmesh,a42 |\
+ 	openmesh,a62 |\
+ 	qxwlan,e2600ac-c1 |\
+-	qxwlan,e2600ac-c2)
++	qxwlan,e2600ac-c2 |\
++	cig,wf610d)
+ 		ath10kcal_extract "0:ART" 4096 12064
+ 		;;
+ 	engenius,ens620ext)
+@@ -238,7 +239,8 @@ case "$FIRMWARE" in
+ 	openmesh,a42 |\
+ 	openmesh,a62 |\
+ 	qxwlan,e2600ac-c1 |\
+-	qxwlan,e2600ac-c2)
++	qxwlan,e2600ac-c2 |\
++	cig,wf610d)
+ 		ath10kcal_extract "0:ART" 20480 12064
+ 		;;
+ 	engenius,ens620ext)
+@@ -266,6 +268,7 @@ case "$FIRMWARE" in
+ 		;;
+ 	esac
+ 	;;
++
+ *)
+ 	exit 1
+ 	;;
+diff --git a/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh b/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
+index f8be25d8ff..29c4b50d6a 100644
+--- a/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
++++ b/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
+@@ -1,3 +1,4 @@
++. /lib/functions/system.sh
+ PART_NAME=firmware
+ REQUIRE_IMAGE_METADATA=1
+ 
+@@ -44,6 +45,20 @@ zyxel_do_upgrade() {
+ 	fi
+ }
+ 
++cig_do_upgrade() {
++	local tar_file="$1"
++
++	local board_dir=$(tar tf $tar_file | grep -m 1 '^sysupgrade-.*/$')
++	board_dir=${board_dir%/}
++	local dev=$(find_mtd_chardev "0:HLOS")
++
++	tar Oxf $tar_file ${board_dir}/kernel | mtd write - ${dev}
++
++	tar Oxf $tar_file ${board_dir}/root | mtd write - rootfs
++}
++
++
++
+ platform_do_upgrade() {
+ 	case "$(board_name)" in
+ 	8dev,jalapeno |\
+@@ -82,6 +97,10 @@ platform_do_upgrade() {
+ 	zyxel,nbg6617)
+ 		zyxel_do_upgrade "$1"
+ 		;;
++	cig,wf610d)
++		cig_do_upgrade "$1"
++		;;
++
+ 	*)
+ 		default_do_upgrade "$1"
+ 		;;
+diff --git a/target/linux/ipq40xx/config-4.14 b/target/linux/ipq40xx/config-4.14
+index c72e693206..46e10475d2 100644
+--- a/target/linux/ipq40xx/config-4.14
++++ b/target/linux/ipq40xx/config-4.14
+@@ -187,7 +187,7 @@ CONFIG_GPIOLIB_IRQCHIP=y
+ CONFIG_GPIO_74X164=y
+ CONFIG_GPIO_SYSFS=y
+ CONFIG_GPIO_WATCHDOG=y
+-# CONFIG_GPIO_WATCHDOG_ARCH_INITCALL is not set
++CONFIG_GPIO_WATCHDOG_ARCH_INITCALL=y
+ # CONFIG_GRO_CELLS is not set
+ CONFIG_HANDLE_DOMAIN_IRQ=y
+ CONFIG_HARDEN_BRANCH_PREDICTOR=y
+@@ -251,6 +251,8 @@ CONFIG_I2C_CHARDEV=y
+ CONFIG_I2C_HELPER_AUTO=y
+ CONFIG_I2C_QUP=y
+ CONFIG_INITRAMFS_SOURCE=""
++CONFIG_INPUT=y
++CONFIG_INPUT_KEYBOARD=y
```

Comment:
do we really have a keyboard ?

Reply:
We need it for gpio key

Reply:
that is why we have gpio-button-hotplug. please use that instead